test/serialization.spec.ts

```diff
@@ -0,0 +1,73 @@
+import {HyperFormula} from '../src'
+import {CellValueDetailedType} from '../src/Cell'
+import {adr} from './testUtils'
+
+describe('serialization', () => {
+  it('should not loose sheet information on serialization', () => {
+    const engine1 = HyperFormula.buildFromArray([
+      [1, '2', 'foo', true, '\'1', '33$', '12/01/15']
+    ])
+
+    expect(engine1.getCellSerialized(adr('A1'))).toEqual(1)
+    expect(engine1.getCellValueFormat(adr('A1'))).toEqual(undefined)
+    expect(engine1.getCellValueDetailedType(adr('A1'))).toEqual(CellValueDetailedType.NUMBER_RAW)
+
+    expect(engine1.getCellSerialized(adr('B1'))).toEqual('2')
+    expect(engine1.getCellValueFormat(adr('B1'))).toEqual(undefined)
+    expect(engine1.getCellValueDetailedType(adr('B1'))).toEqual(CellValueDetailedType.NUMBER_RAW)
+
+    expect(engine1.getCellSerialized(adr('C1'))).toEqual('foo')
+    expect(engine1.getCellValueFormat(adr('C1'))).toEqual(undefined)
+    expect(engine1.getCellValueDetailedType(adr('C1'))).toEqual(CellValueDetailedType.STRING)
+
+    expect(engine1.getCellSerialized(adr('D1'))).toEqual(true)
+    expect(engine1.getCellValueFormat(adr('D1'))).toEqual(undefined)
+    expect(engine1.getCellValueDetailedType(adr('D1'))).toEqual(CellValueDetailedType.BOOLEAN)
+
+    expect(engine1.getCellSerialized(adr('E1'))).toEqual('\'1')
+    expect(engine1.getCellValueFormat(adr('E1'))).toEqual(undefined)
+    expect(engine1.getCellValueDetailedType(adr('E1'))).toEqual(CellValueDetailedType.STRING)
+
+    expect(engine1.getCellSerialized(adr('F1'))).toEqual('33$')
+    expect(engine1.getCellValueFormat(adr('F1'))).toEqual('$')
+    expect(engine1.getCellValueDetailedType(adr('F1'))).toEqual(CellValueDetailedType.NUMBER_CURRENCY)
+
+    expect(engine1.getCellSerialized(adr('G1'))).toEqual('12/01/15')
+    expect(engine1.getCellValueFormat(adr('G1'))).toEqual(undefined)
```

Comment:
Can we add a note that this is wrong? With link to https://github.com/handsontable/hyperformula/issues/626

Reply:
riiiight